File: inflow-backend-main/src/main/java/com/john/inflow/service/ProductService.java
package com.john.inflow.service;

import com.john.inflow.dto.request.CreateProductRequest;
import com.john.inflow.dto.request.UpdateProductRequest;
import com.john.inflow.dto.response.ProductResponse;

import java.util.List;

public interface ProductService {
    ProductResponse create(CreateProductRequest request);
    ProductResponse getById(Integer id);
    List<ProductResponse> getAll();
    ProductResponse update(Integer id, UpdateProductRequest request);
    void delete(Integer id);
}

File: src/services/product.service.ts
import api from "./api";
import {
  Product,
  ProductDetail,
  ProductStats,
  CreateProductRequest,
  UpdateProductRequest,
  ProductFilterParams,
} from "@/types/product.types";
import { ApiResponse, PaginatedResponse } from "@/types/api.types";

export const productService = {
  getAll: (params?: ProductFilterParams) =>
    api.get<PaginatedResponse<Product>>("/products", { params }),

  getById: (id: number) =>
    api.get<ApiResponse<Product>>(`/products/${id}`),

  getDetail: (id: number) =>
    api.get<ApiResponse<ProductDetail>>(`/products/${id}/detail`),

  getStats: () =>
    api.get<ApiResponse<ProductStats>>("/products/summary"),

  create: (data: CreateProductRequest) =>
    api.post<ApiResponse<Product>>("/products", data),

  update: (id: number, data: UpdateProductRequest) =>
    api.put<ApiResponse<Product>>(`/products/${id}`, data),

  deactivate: (id: number) =>
    api.post<void>(`/products/${id}/deactivate`),

  delete: (id: number) =>
    api.delete<ApiResponse<void>>(`/products/${id}`),
};
